ext/mysqli/tests/mysqli_constants.phpt

```diff
@@ -9,229 +9,216 @@ require_once('skipifconnectfailure.inc');
+}
+
+/* First introduced in MySQL 6.0, backported to MySQL 5.5 */
+if ($version >= 50500 || $IS_MYSQLND) {
+    $expected_constants['MYSQLI_SERVER_QUERY_WAS_SLOW'] = true;
+}
+
+$expected_constants['MYSQLI_CLIENT_SSL_VERIFY_SERVER_CERT'] = true;
+if ($IS_MYSQLND) {
+    $expected_constants['MYSQLI_CLIENT_SSL_DONT_VERIFY_SERVER_CERT'] = true;
+}
+
+/* First introduced in MySQL 6.0, backported to MySQL 5.5 */
+if ($version >= 50606 || $IS_MYSQLND) {
+    $expected_constants['MYSQLI_SERVER_PUBLIC_KEY'] = true;
+}
+
+$expected_constants = array_merge($expected_constants, array(
+    "MYSQLI_TYPE_NEWDECIMAL"	=> true,
+    "MYSQLI_TYPE_BIT"			=> true,
+));
+
+$expected_constants['MYSQLI_NO_DEFAULT_VALUE_FLAG'] = true;
+
+$expected_constants = array_merge($expected_constants, array(
+    "MYSQLI_STMT_ATTR_CURSOR_TYPE"		=> true,
+    "MYSQLI_CURSOR_TYPE_NO_CURSOR"		=> true,
+    "MYSQLI_CURSOR_TYPE_READ_ONLY"		=> true,
+    "MYSQLI_CURSOR_TYPE_FOR_UPDATE"		=> true,
+    "MYSQLI_CURSOR_TYPE_SCROLLABLE"		=> true,
+));
+
+$expected_constants = array_merge($expected_constants, array(
+    "MYSQLI_STMT_ATTR_PREFETCH_ROWS"	=> true,
+));
+
+if ($version < 80000 || $version >= 100000 || $IS_MYSQLND) {
+    $expected_constants['MYSQLI_OPT_SSL_VERIFY_SERVER_CERT'] = true;
+}
+
+/* pretty dump test, but that is the best way to mimic mysql.c */
+$expected_constants["MYSQLI_DATA_TRUNCATED"] = true;
+
+if ($IS_MYSQLND || (!$IS_MYSQLND && ($version > 50610))) {
+    /* could be that MySQL/libmysql 5.6.9 had the flag already but it was no stable release */
+    $expected_constants["MYSQLI_OPT_CAN_HANDLE_EXPIRED_PASSWORDS"] = true;
+    $expected_constants["MYSQLI_CLIENT_CAN_HANDLE_EXPIRED_PASSWORDS"] = true;
+}
+
+if ($IS_MYSQLND) {
+    $expected_constants["MYSQLI_TYPE_JSON"]	= true;
+}
+
+if ($version > 80210 || $IS_MYSQLND) {
+    $expected_constants['MYSQLI_OPT_LOAD_DATA_LOCAL_DIR'] = true;
+}
+
+$unexpected_constants = array();
+
+foreach ($constants as $group => $consts) {
+    foreach ($consts as $name => $value) {
+        if (stristr($name, 'mysqli')) {
+            $name = strtoupper($name);
+            if (isset($expected_constants[$name])) {
+                unset($expected_constants[$name]);
+            } else {
+                $unexpected_constants[$name] = $name;
             }
         }
     }
+}
 
-    if (!empty($unexpected_constants)) {
-        printf("Dumping list of unexpected constants\n");
-        var_dump($unexpected_constants);
-    }
+if (!empty($unexpected_constants)) {
+    printf("Dumping list of unexpected constants\n");
+    var_dump($unexpected_constants);
+}
 
-    if (!empty($expected_constants)) {
-        printf("Dumping list of missing constants\n");
-        var_dump($expected_constants);
-    }
+if (!empty($expected_constants)) {
+    printf("Dumping list of missing constants\n");
+    var_dump($expected_constants);
+}
 
-    print "done!";
+print "done!";
+?>
+--CLEAN--
+<?php
+require_once "clean_table.inc";
```

Comment:
Why does this test even need a table in the first place? Can we just avoid creating it?